lib/Backend/GlobOptBailOut.cpp

```diff
@@ -884,10 +884,22 @@ void GlobOpt::EndTrackCall(IR::Instr* instr)
 }
 
 void
-GlobOpt::FillBailOutInfo(BasicBlock *block, BailOutInfo * bailOutInfo)
+GlobOpt::FillBailOutInfo(BasicBlock *block, BailOutInfo * bailOutInfo, IR::Instr * instr)
 {
     AssertMsg(!this->isCallHelper, "Bail out can't be inserted the middle of CallHelper sequence");
 
+    if (instr && this->isRecursiveCallOnLandingPad)
```

Comment:
nit: I would add an overload of `FillBailOutInfo(BasicBlock*, IR::Instr*)` that does this extra check and then calls `FillBailOutInfo(block, instr->GetBailoutInfo())`

Reply:
Sure, makes sense